File: examples/kotlin/src/main/java/com/examples/overloads/overload.kt
package com.examples.overloads

import alias.*



/**
 * An example for with overloaded methods.
 */
open class OverloadedFunctions
    internal constructor(_id: Long) : AutoCloseable {
    companion object {
        init {
          System.loadLibrary("wrapper_jni");
        }
        /**
         * Overload constructor
         * 
         */
        protected fun construct_helper(): Long {
            
            val id = jConstructor()
            return id
        }
        @JvmStatic
        private external fun jConstructor(): Long


    }
    
    protected var id = _id
    
    open fun getObjId(): Long {
        if(id == 0L) {
            throw RuntimeException("Object is not allocated")
        }
        return id;
    }
    /**
     * Overload constructor
     * 
     */
    constructor(): this(construct_helper()) {
      //jSet_this(id, this)
    }
    
    
    /**
     * Concatenate with two strings.
     * 
     */
    fun concatenate(first: String, second: String): String {
        
        
        
        val result = jConcatenate(getObjId(), first, second)
        
        return result
    }

    /**
     * Concatenate with three strings.
     * 
     */
    fun concatenate(first: String, second: String, third: String): String {
        
        
        
        
        val result = jConcatenate_1(getObjId(), first, second, third)
        
        return result
    }
    override fun close() {
    	if (id != 0L) {
    		jFinalize(id)
    		id = 0L
    	}
    }
    /**
    * Finalize and deletes the object
    */
    protected fun finalize() {
    	close()
    }
    ///// External wrapper functions ////////////
    private external fun jConcatenate(id: Long, first: String, second: String): String
    private external fun jConcatenate_1(id: Long, first: String, second: String, third: String): String
    private external fun jSet_this(id: Long, self: Any): Void
    private external fun jFinalize(id: Long): Void
}

File: examples/primitives/swift/src/overloads/overload.swift
import Foundation

import CWrapper

/**
 * An example for with overloaded methods.
 */
public class OverloadedFunctions  {
    
    public let cself : UnsafeMutableRawPointer
    public let owner : Bool

    // internal main initializer
    internal required init(_ _cself: UnsafeMutableRawPointer , _ _owner: Bool = false) {
      self.cself = _cself
      self.owner = _owner
    }
    deinit {
      if owner {
        release_OverloadedFunctions(cself)
      }
    }
    
    /**
     * Overload constructor
     * 
     */
    public convenience init() {

        self.init(create_OverloadedFunctions(), true)
    }
    
    /**
     * Concatenate with two strings.
     * 
     */
    public func concatenate(first: String, second: String) -> String {

        let swift_to_sc_first = strdup(first)!
        let swift_to_sc_second = strdup(second)!
        let result = _func_OverloadedFunctions_concatenate(cself, swift_to_sc_first, swift_to_sc_second);
        let sc_to_swift_result = String(cString: result)
        defer{
          result.deallocate()
        }
        return sc_to_swift_result;
    }
    /**
     * Concatenate with three strings.
     * 
     */
    public func concatenate(first: String, second: String, third: String) -> String {

        let swift_to_sc_first = strdup(first)!
        let swift_to_sc_second = strdup(second)!
        let swift_to_sc_third = strdup(third)!
        let result = _func_OverloadedFunctions_concatenate_1(cself, swift_to_sc_first, swift_to_sc_second, swift_to_sc_third);
        let sc_to_swift_result = String(cString: result)
        defer{
          result.deallocate()
        }
        return sc_to_swift_result;
    }
}